app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+class CommentsController < ApplicationController
+  include CommentsHelper
+  before_action :authenticate_user!
+
+  def create
+    @comment = @commentable.comments.new(comment_params)
+    @comment.user = current_user
+    respond_to do |format|
+      if @comment.save
+        format.html { comments_list }
+      else
+        format.json { comment_error }
+      end
+    end
+  end
+
+  def destroy
+    @comment = Comment.find(params[:id])
+    @commentable = @comment.commentable
+    if rights_to_remove?(@comment)
```

Comment:
ideally this method should look like `comment.can_delete?(current_user)` ... instead of this helper
same for edit, etc